### DISCUSSION TITLE
Magento importer supporting multiselect items

### DISCUSSION THREAD

**Discussion by seansan** [OpenMage/magento-lts#2100] [Ideas]
Currently when we import magento data and have a multiselect item (say colors)

When we want to mark a product as having red, green and yellow

In our import sheets we need to add 3 lines!

(I think) in previous versions one could import the multiselect data using a separator like here: https://magento.stackexchange.com/questions/12368/how-import-product-multiselect-attributes-with-profiler?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

This does no longer seem to work

Request: we would really like to import data for multiselect items on 1 line using a separator

(noted for future reference or anyone who feels like updating the code)

**Comment by Quazz** [comment]
I don't personally use magento's import (I use magmi). Could you post an example line that fails?

Afaik in 1.9.3 Magento changed multiselect from varchar to text type. Maybe other changes too that cause this behavior. Or maybe they just never updated the importer to handle the changes.

**Comment by seansan** [comment]
​​
​I read some​where that it was possible in the past using a pipe as
separator (val1|val2|val3) & also using comma separators
([data][space][comma][space][data][space])

But currently the only way is to add an extra line to import sheets => and
this is unworkable

best would be to support both methods above:

so importer finds an import field
then detects it is a multiselect field
then check for separators
then loops over the choices using the separators above
and multi selects them

in addition one could add a simple quick win where there is support for a
leading + and -, where the + adds the selection "on top" of the current
multiselect selection and minus removes it.

for now?
a simple code update to support  using a pipe as separator would be a GREAT
WIN

Does this help?

thanks


https://magento.stackexchange.com/questions/12368/how-import-product-multiselect-attributes-with-profiler

https://magento.stackexchange.com/questions/212875/magento-2-import-multiple-select-attribute-values-using-csv

https://magento.stackexchange.com/questions/100099/csv-import-with-multi-select-attributes

**Comment by tomekjordan** [comment]
I doubt anybody uses built in VERY LIMITED Magento importer when you have much better tools like Store Manager [https://www.mag-manager.com/](https://www.mag-manager.com/)

**Comment by seansan** [comment]
​​
​Thanks for that Tomek

however people do use it

​

**Comment by Flyingmana** [comment]
most people I know use https://github.com/avstudnitz/AvS_FastSimpleImport

But generally yes, we should have proper support, on the other side this is a very complex topic, and I think it would be healthier for the project, to outsource this to the external solutions and recommend them to be used. This way we also dont need to care about compatibility to Magento2, as they will likely have a Magento2 implementation ready
